uvicorn/protocols/websockets/websockets_impl.py

```diff
@@ -209,28 +243,32 @@ async def run_asgi(self):
                 await self.handshake_completed_event.wait()
                 self.transport.close()
 
-    async def asgi_send(self, message):
+    async def asgi_send(self, message: ASGISendEvent) -> None:
         message_type = message["type"]
 
         if not self.handshake_started_event.is_set():
             if message_type == "websocket.accept":
+                message = cast(WebSocketAcceptEvent, message)
                 self.logger.info(
                     '%s - "WebSocket %s" [accepted]',
                     self.scope["client"],
                     self.scope["path"],
                 )
                 self.initial_response = None
-                self.accepted_subprotocol = message.get("subprotocol")
+                self.accepted_subprotocol = message.get(  # type: ignore[assignment]
+                    "subprotocol"
+                )
```

Comment:
Just curious - why does this need an ignore? What error do we get without it?

Reply:
Because `accepted_subprotocol` was a `websockets.Subprotocol` type, but I've replaced it by simple `str` and remove the ignore.

Reply:
I had to readd the `websockets.Subprotocol` because the parent class uses it.